assets/plugins/hash/hash.js

```diff
@@ -0,0 +1,237 @@
+                    this.removeLayer(targetMarker);
+
+                    // add marker to cluster
+                    this.markers.addLayer(targetMarker);
+                }
+
+                // clear hash and set new location hash
+                this._clearHash();
+                this._setHash();
+            }
+        }
+    },
+
+    _setLocation: function (hash) {
+        var args = hash ? hash.split(',') : [];
+
+        // set up location properties
+        var hashObj = {
+            zoom: Number(args[0]),
+            lat: Number(args[1]),
+            lng: Number(args[2])
+        };
+
+        // select center position
+        var center;
+        if (!isNaN(hashObj.lat) && !isNaN(hashObj.lng)) {
+            center = new L.LatLng(hashObj.lat, hashObj.lng);
+        } else {
+            center = this.getCenter();
+        }
+
+        // select zoom
+        var zoom;
+        if (!isNaN(hashObj.zoom)) {
+            zoom = hashObj.zoom;
+        }
+
+        // update map
+        this.setView(center, zoom, {
+            animate: false
+        });
+    },
+
+    _setUser: function (id) {
+        // check for existing user entry and get marker
+        var targetEntry = this._getUserEntry(id);
+        var targetMarker = targetEntry ? this._getMarker(targetEntry.id) : false;
+        if (targetMarker) {
+
+            // center map around target marker, use medium zoom
+            this.setView([targetEntry.latitude, targetEntry.longitude], 6, {
+                animate: false
+            });
+
+            // transfer from cluster to map (to allow for popup)
+            if (this.markers.hasLayer(targetMarker)) {
+
+                // remove marker from clusters
+                this.markers.removeLayer(targetMarker);
+
+                // add marker to map and open popup
+                targetMarker.addTo(this);
+            }
+
+            // open popup
+            targetMarker.openPopup();
+        }
+    },
+
+    _getUserEntry: function (userID) {
+        // select user data entry by given id
+        var item = directory.filter(function (entry) {
+            return entry.id === userID;
+        }, this);
+        if (item.length) {
+            return item[0];
+        }
+        return false;
+    },
+
+    _getMarker: function (id) {
+        var sources = [this, this.markers]; // both map markers and cluster markers
+        var targetLayer = false;
+
+        // loop through map layers and cluster layers to find target marker
+        sources.forEach(function (source) {
+            if (!targetLayer && typeof source.eachLayer === "function") {
+                source.eachLayer(function (layer) {
+                    if (layer.userID == id && layer instanceof L.Marker) {
+                        targetLayer = layer;
+                    }
+                });
+            }
+        });
+        return targetLayer;
+    },
+
+    _getHash: function () {
+        var hash = window.location.hash.slice(1);
```

Comment:
Ist .hash immer defined? Oder nur mit # in der url?

Reply:
Ist immer definied. Kommt leer zurück, wenn kein Hash an der URL klebt.